poseidon/poseidonMonitor/NorthBoundControllerAbstraction/proxy/bcf/bcf.py

```diff
@@ -141,11 +144,16 @@ def get_span_fabric(
         if span_name:
             span_fabric_resource = ''.join(
                 [span_fabric_resource, '[name="%s"]' % span_name])
-        if self.interface_group:
+        if interface_group:
             span_fabric_resource = ''.join(
-                [span_fabric_resource, '[dest-interface-group="%s"]' % self.interface_group])
+                [span_fabric_resource, '[dest-interface-group="%s"]' % interface_group])
         r = self.get_resource(span_fabric_resource)
-        retval = BcfProxy.parse_json(r)[0]
+        spanArray = BcfProxy.parse_json(r)
+        print("{0} ; {1} ".format(span_name, interface_group))
+        if len(spanArray) == 0:
+            raise EnvironmentError("A span fabric with the configured combination of name: {0} and interface group: {1} could not be"
```

Comment:
wasn't able to actually trigger this. are you sure spanArray returns an empty array and that it's actually an object, and not a string or something?

Reply:
here is a gist showing the request being run in that scenario:
https://gist.github.com/rashley-iqt/1a7657b360f1d5ea9d282e40174311aa

if you fill in the server, username and password it will return an empty array.

Reply:
i guess what I meant was when actually having Poseidon running, if you give it incorrect values or no values, it doesn't actually trigger.